beetsplug/importhistory.py

```diff
@@ -0,0 +1,157 @@
+"""Adds a `source_path` attribute to imported albums indicating from what path
+the album was imported from. Also suggests removing that source path in case
+you've removed the album from the library.
+
+"""
+
+import os
+import re
+from shutil import rmtree
+
+from beets.plugins import BeetsPlugin
+from beets.ui import colorize as colorize_text
+from beets.ui import input_options, input_yn
+
+
+class ImportHistPlugin(BeetsPlugin):
+    """Main plugin class."""
+
+    def __init__(self):
+        """Initialize the plugin and read configuration."""
+        super(ImportHistPlugin, self).__init__()
+        self.config.add(
+            {
+                "auto": True,
+            }
+        )
+        if self.config["auto"]:
+            self.import_stages = [self.import_stage]
+            self.register_listener("item_removed", self.suggest_removal)
+        # In order to stop suggestions for certain albums in
+        # self.suggest_removal, we initialize an empty list of mb_albumids that
+        # these will be ignored in future runs
+        self.stop_suggestions_for_albums = []
+
+    def import_stage(self, _, task):
+        """Event handler for albums import finished."""
+        # first, we sort the items using a dictionary and a list of paths for
+        # every mbid
+        for item in task.imported_items():
+            item["source_path"] = item.path
+            item.try_sync(write=True, move=False)
+
+    def suggest_removal(self, item):
+        """Prompts the user to delete the original path the item was imported from."""
+        if "source_path" not in item:
+            try:
+                self._log.warn(
+                    "Item without a source_path was found: {0.title} by {0.artist}",
+                    item,
+                )
+            except UnicodeDecodeError:
+                self._log.warn(
+                    "Item without a source_path was found: mb_trackid: {0.mb_trackid}",
+                    item,
+                )
+            return
+        if (
+            not item["source_path"]
+            or item.mb_albumid in self.stop_suggestions_for_albums
+        ):
+            return
+        if os.path.isdir(item["source_path"]):
```

Comment:
Also, this might be a little more readable as `item.source_path` instead of `item["source_path"]`.

Reply:
> Also, this might be a little more readable as `item.source_path` instead of `item["source_path"]`.

Are you sure these the former attribute exists?

Reply:
yes, beets library object's flexible attributes are accessible both ways. As long as it was set to the Item object, it will be accessible like that.

Reply:
Yep, these ways of accessing the data work identically!